database/worker.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package database
+
+import (
+	"database/sql"
+
+	"github.com/go-vela/types/library"
+	"github.com/lib/pq"
+)
+
+// Worker is the database representation of a worker.
+type Worker struct {
+	ID            sql.NullInt64  `sql:"id"`
+	Hostname      sql.NullString `sql:"hostname"`
+	Address       sql.NullString `sql:"address"`
+	Routes        pq.StringArray `sql:"routes"`
+	Active        sql.NullBool   `sql:"active"`
+	LastCheckedIn sql.NullTime   `sql:"last_checked_in"`
+}
+
+// Nullify ensures the valid flag for
+// the sql.Null types are properly set.
+//
+// When a field within the Build type is the zero
+// value for the field, the valid flag is set to
+// false causing it to be NULL in the database.
+func (w *Worker) Nullify() *Worker {
+	if w == nil {
+		return nil
+	}
+
+	// check if the ID field should be false
+	if w.ID.Int64 == 0 {
+		w.ID.Valid = false
+	}
+
+	// check if the Hostname field should be false
+	if len(w.Hostname.String) == 0 {
+		w.Hostname.Valid = false
+	}
+
+	// check if the Address field should be false
+	if len(w.Address.String) == 0 {
+		w.Hostname.Valid = false
+	}
+
+	return w
+}
+
+// ToLibrary converts the Worker type
+// to a library Worker type.
+func (w *Worker) ToLibrary() *library.Worker {
+	worker := new(library.Worker)
+
+	worker.SetID(w.ID.Int64)
+	worker.SetHostname(w.Hostname.String)
+	worker.SetAddress(w.Address.String)
+	worker.SetRoutes(w.Routes)
+	worker.SetActive(w.Active.Bool)
+	worker.SetLastCheckedIn(w.LastCheckedIn.Time)
+	return worker
+}
+
+// Validate verifies the necessary fields for
+// the Repo type are populated correctly.
+func (w *Worker) Validate() error {
+	// verify the Host field is populated
+	if len(w.Hostname.String) == 0 {
+		return ErrEmptyRepoHash
+	}
+
+	// verify the Address field is populated
+	if len(w.Address.String) == 0 {
+		return ErrEmptyRepoOrg
```

Comment:
I think we might want to change this error message 👍 

Typically we create unique error messages per field per struct.

Currently, it looks like we're referencing this error:

https://github.com/go-vela/types/blob/3e7f85be5d18e6fe39e992946a2b7eaa903800d7/database/repo.go#L27-L29